code/datums/ai/_ai_controller.dm

```diff
@@ -130,6 +132,54 @@ multiple modular subtrees with behaviors
 	RegisterSignal(pawn, COMSIG_MOB_STATCHANGE, PROC_REF(on_stat_changed))
 	RegisterSignal(pawn, COMSIG_MOB_LOGIN, PROC_REF(on_sentience_gained))
 	RegisterSignal(pawn, COMSIG_QDELETING, PROC_REF(on_pawn_qdeleted))
+	RegisterSignal(pawn, COMSIG_GRID_UPDATED, PROC_REF(update_grid))
```

Comment:
The grids we overlap with depends not on the cell we are in but on the position we have in the world. You need to use Moved() for this (its signal)